drivers/platform/xilinx/i2c.c

```diff
@@ -55,10 +55,124 @@
 #include "i2c.h"
 #include "i2c_extra.h"
 
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/** Table of existing BSP driver handlers */
+#ifdef XPAR_XIIC_NUM_INSTANCES
+/**
+ * @struct pl_table_item
+ * @brief Table item for PL I2C cores already started.
+ */
+struct pl_table_item {
+	/** Instance the BSP driver uses for this core. */
+	XIic *instance;
+	/** Core ID in BSP. */
+	uint32_t device_id;
+	/** Number of ADI driver instances running on this core. */
+	uint32_t inst_no;
+};
+/** Dynamically allocated table for PL I2C BSP driver instances. */
+static struct pl_table_item *pl_table = NULL;
```

Comment:
I think you can use a list and the associated functions to handle this.
Also using a list functions used for find and for allocation will not be needed

Reply:
Do you mean the FIFO utils? Or which list?

Reply:
https://github.com/analogdevicesinc/no-OS/pull/493#pullrequestreview-368703841

Reply:
I would like to do that, but I will try and do it after it has been merged to master.

Reply:
Fine for me